### PR TITLE
Continue polling engine_exchangeConfiguration after the merge completes

### LINKED ISSUE #5324: [Merge] Continue calling `engine_exchangeConfig` after merge completes
### Description
The spec changed to say Els should complain if they don't get regular calls to exchangeConfig so we need to carry on calling even after the merge completes as the EL may not yet think the merge is complete.

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
From the execution engine spec: EL clients SHOULD log an error if they don't receive a call to engine_exchangeConfiguration at least once every 120 seconds. This doesn't include a provision for stopping polling after the merge completes, and the EL may not have the same view of the merge status as we do due to different syncing stages.  So we need to keep polling even after we see the merge as complete.

## Fixed Issue(s)
fixes #5324 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    when(executionEngine.exchangeTransitionConfiguration(localTransitionConfiguration))
+        .thenReturn(SafeFuture.completedFuture(wrongRemoteConfig));
+
+    asyncRunner.executeQueuedActions();
+
+    verify(eventLogger)
+        .transitionConfigurationTtdTbhMismatch(
+            localTransitionConfiguration.toString(), wrongRemoteConfig.toString());
+
+    // wrong TBH
```
Comment (rolfyone) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java]: TBH being terminal block hash?
not `to be honest`....

Comment (ajsutton) [reply ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java]: I think so - I just copied that bit of code. :) But will expand it out.

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    when(executionEngine.exchangeTransitionConfiguration(localTransitionConfiguration))
+        .thenReturn(SafeFuture.completedFuture(wrongRemoteConfig));
+
+    asyncRunner.executeQueuedActions();
+
+    verify(eventLogger)
+        .transitionConfigurationTtdTbhMismatch(
+            localTransitionConfiguration.toString(), wrongRemoteConfig.toString());
+
+    // remote TBH TBN inconsistency
```
Comment (rolfyone) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java]: these do sound like we're lumping a stack of tests into one... should they be separate?

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                    localTransitionConfiguration.getTerminalTotalDifficulty(),
+                    localTransitionConfiguration.getTerminalBlockHash(),
+                    dataStructureUtil.randomUInt64())));
+
+    assertThat(mergeTransitionConfigCheck.start()).isCompleted();
+  }
+
+  @Test
+  void shouldReportWrongTotalTerminalDifficulty() {
+    // wrong terminal total difficulty
```
Comment (rolfyone) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java]: now that we've got the test names all pretty can eliminate the comments, but thats admittedly a super nit...

**Review by rolfyone — APPROVED** [review]
LGTM

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java (hunk 3 above)
Comment (ajsutton) [reply ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java]: Yeah makes sense.

**Comment by ajsutton** [comment]
FYI @jflo

**Pull request merged**